Build failure: fahclient

### Steps To Reproduce

Steps to reproduce the behavior:
1. `nix-shell -p fahclient`

### Build log

```
In file included from src/cbang/openssl/BigNum.cpp:33:
src/cbang/openssl/BigNum.h:55:14: error: ‘uint64_t’ has not been declared
   55 |     void set(uint64_t x);
      |              ^~~~~~~~
g++ -o build/cbang/openssl/Cipher.o -c -faligned-new -std=c++14 -fsigned-char -ffunction-sections -fdata-sections -O3 -funroll-loops -fno-pie -fPIC -DNDEBUG -D_REENTRANT -DUSING_CBANG -Iinclude -Isrc -Isrc/boost src/cbang/openssl/Cipher.cpp
g++ -o build/cbang/openssl/Digest.o -c -faligned-new -std=c++14 -fsigned-char -ffunction-sections -fdata-sections -O3 -funroll-loops -fno-pie -fPIC -DNDEBUG -D_REENTRANT -DUSING_CBANG -Iinclude -Isrc -Isrc/boost src/cbang/openssl/Digest.cpp
src/cbang/openssl/BigNum.cpp:62:6: error: no declaration matches ‘void cb::BigNum::set(uint64_t)’
   62 | void BigNum::set(uint64_t x) {BN_set_word(bn, x);}
      |      ^~~~~~
src/cbang/openssl/BigNum.h:55:10: note: candidate is: ‘void cb::BigNum::set(int)’
   55 |     void set(uint64_t x);
      |          ^~~
src/cbang/openssl/BigNum.h:40:9: note: ‘class cb::BigNum’ defined here
   40 |   class BigNum {
      |         ^~~~~~
scons: *** [build/cbang/openssl/BigNum.o] Error 1
scons: building terminated because of errors.
/nix/store/v5irq7wvkr7kih0hhnch5nnv2dcq8c4f-stdenv-linux/setup: line 131: pop_var_context: head of shell_variables not a function context
error: builder for '/nix/store/ksh2xpdb2dqmjxdl8zl99w9gzg6ci1hd-fah-client-8.1.18.drv' failed with exit code 2;
       last 10 log lines:
       >       |      ^~~~~~
       > src/cbang/openssl/BigNum.h:55:10: note: candidate is: ‘void cb::BigNum::set(int)’
       >    55 |     void set(uint64_t x);
       >       |          ^~~
       > src/cbang/openssl/BigNum.h:40:9: note: ‘class cb::BigNum’ defined here
       >    40 |   class BigNum {
       >       |         ^~~~~~
       > scons: *** [build/cbang/openssl/BigNum.o] Error 1
       > scons: building terminated because of errors.
       > /nix/store/v5irq7wvkr7kih0hhnch5nnv2dcq8c4f-stdenv-linux/setup: line 131: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/ksh2xpdb2dqmjxdl8zl99w9gzg6ci1hd-fah-client-8.1.18.drv'.
error: 1 dependencies of derivation '/nix/store/xwgr68gmpfma51pc2zm3q4x4w3bnxmg8-fah-client-8.1.18-usr-target.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xny4jkjqiikkn5ppi25gy05z92acklwd-fah-client-8.1.18-fhs.drv' failed to build
error: 1 dependencies of derivation '/nix/store/736zwliyn9s990k40xx2vlnpix2jyy33-fah-client-8.1.18-bwrap.drv' failed to build
error: 1 dependencies of derivation '/nix/store/mmfqqjhdf5mn17nmx9ah83zvbdsvhf38-fah-client-8.1.18.drv' failed to build
```

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@zimbatm 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/v8p3m2vgkcd3ykrdylm5vi9alrxbwhip-nix-info
copying path '/nix/store/v8p3m2vgkcd3ykrdylm5vi9alrxbwhip-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.0-10-amd64, Debian GNU/Linux, 12 (bookworm), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/store/7ddzmp96d5m8j4fmh057akxi6fml7jm7-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/science/misc/foldingathome/client.nix
+++ b/pkgs/applications/science/misc/foldingathome/client.nix
@@ -8,13 +8,13 @@
 , extraPkgs ? [ ]
 }:
 let
-  version = "8.1.18";
+  version = "8.3.1";
 
   cbangSrc = fetchFromGitHub {
     owner = "cauldrondevelopmentllc";
     repo = "cbang";
     rev = "bastet-v${version}";
-    hash = "sha256-G0rknVmZiyC4sRTOowFjf7EQ5peGf+HLPPcLWXXFlX4=";
+    hash = "sha256-cuyfJG5aDJ6e2SllxwKTViG0j8FWHvjcTaaBBtkgEdU=";
   };
 
   fah-client = stdenv.mkDerivation {
@@ -25,7 +25,7 @@ let
       owner = "FoldingAtHome";
       repo = "fah-client-bastet";
       rev = "v${version}";
-      hash = "sha256-IgT/5NqCwN8N8OObjtASuT4IRb2EN4bdixxUdjiyddI=";
+      hash = "sha256-Ztc2im4Xmk8f6GotGRgA5zDkcyQFnodUvroJVl+ApT4=";
     };
 
     nativeBuildInputs = [ scons ];
@@ -45,14 +45,16 @@ let
     installPhase = ''
       runHook preInstall
 
-      mkdir -p $out/{bin,share/applications,share/feh-client}
+      mkdir -p $out/{bin,share/feh-client}
 
       cp fah-client $out/bin/fah-client
 
-      cp install/lin/fah-client.desktop $out/share/applications/
-      cp -r images $out/share/feh-client/
-
-      sed -e "s|Icon=.*|Icon=$out/share/feh-client/images/fahlogo.png|g" -i $out/share/applications/fah-client.desktop
+      mkdir -p $out/share/applications
+      cp install/lin/fah-client.desktop.in $out/share/applications/fah-client.desktop
+      sed \
+        -e "s|Icon=.*|Icon=$out/share/feh-client/images/fahlogo.png|g" \
+        -e "s|%(PACKAGE_URL)s|https://github.com/FoldingAtHome/fah-client-bastet|g" \
+        -i $out/share/applications/fah-client.desktop
 
       runHook postInstall
     '';

--- a/pkgs/applications/science/misc/foldingathome/client.nix
+++ b/pkgs/applications/science/misc/foldingathome/client.nix
@@ -49,7 +49,6 @@ let
 
       cp fah-client $out/bin/fah-client
 
-      mkdir -p $out/share/applications
       cp install/lin/fah-client.desktop.in $out/share/applications/fah-client.desktop
       sed \
         -e "s|Icon=.*|Icon=$out/share/feh-client/images/fahlogo.png|g" \

--- a/pkgs/applications/science/misc/foldingathome/client.nix
+++ b/pkgs/applications/science/misc/foldingathome/client.nix
@@ -45,7 +45,7 @@ let
     installPhase = ''
       runHook preInstall
 
-      mkdir -p $out/{bin,share/feh-client}
+      mkdir -p $out/{bin,share/applications}
 
       cp fah-client $out/bin/fah-client
 

